sotodlib/hwp/hwp.py

```diff
@@ -6,429 +6,577 @@
 logger = logging.getLogger(__name__)
 
 
-def func_hwpss_2f4f(x, A2, A4, phi2, phi4):
-    return A2 * np.cos(2 * x + phi2) + A4 * np.cos(4 * x + phi4)
+def extract_hwpss(aman, signal=None, hwp_angle=None,
+                  bin_signal=True, bins=3600,
+                  lin_reg=True, modes=[1, 2, 3, 4, 6, 8],
+                  apply_prefilt=True, prefilt_cutoff=1.0,
+                  mask_flags=True,
```

Comment:
add an option to pass a `RangesMatrix` as a flags keyword. In the future we may not want to use just the glitches flag. 

Example:
```
if flags is None:
    flags = aman.flags.glitches
```